examples/basecoin/types/account.go

```diff
@@ -3,25 +3,60 @@ package types
 import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	"github.com/cosmos/cosmos-sdk/x/auth"
+	crypto "github.com/tendermint/go-crypto"
 )
 
 var _ sdk.Account = (*AppAccount)(nil)
 
+// Custom extensions for this application.  This is just an example of
+// extending auth.BaseAccount with custom fields.
+//
+// This is compatible with the stock auth.AccountStore, since
+// auth.AccountStore uses the flexible go-wire library.
 type AppAccount struct {
 	auth.BaseAccount
-
-	// Custom extensions for this application.  This is just an example of
-	// extending auth.BaseAccount with custom fields.
-	//
-	// This is compatible with the stock auth.AccountStore, since
-	// auth.AccountStore uses the flexible go-wire library.
 	Name string
 }
 
-func (acc AppAccount) GetName() string {
-	return acc.Name
+// nolint
+func (acc AppAccount) GetName() string      { return acc.Name }
+func (acc *AppAccount) SetName(name string) { acc.Name = name }
+
+//___________________________________________________________________________________
+
+// We use GenesisAccount instead of AppAccount for cleaner json input of PubKey
+type GenesisAccount struct {
+	Name     string         `json:"name"`
+	Address  crypto.Address `json:"address"`
+	Coins    sdk.Coins      `json:"coins"`
+	PubKey   cmn.HexBytes   `json:"public_key"`
```

Comment:
I guess this needs to be rebased on develop where PubKey/Sequence are removed ?

Reply:
addressed